elastic/kibana
# [EPM] Use index templates v2

## Summary

Implements the alpha1 / 7.8 parts of https://github.com/elastic/kibana/issues/63950

* handles v1 and v2 prebuilt index templates in the `base` package
* only generates v2 index templates for package datasets

This should not affect how we update packages, as long as we don't use index template composition for generated v2 index templates. 

## How to test this

* Run this PR against our default package registry, with a fresh elasticsearch database (e.g. by restarting `yarn es snapshot`).
* Kibana should start without errors, login should work without errors.
* Verify in the Kibana dev console or with `curl` that after login, `GET _cat/templates` correctly lists all packages from the `base` package that was installed during setup (that is, `event-mappings`, `event-settings`, `logs-mappings`, `logs-settings`, `metrics-mappings` and `metrics-settings`.
---
* Then run https://github.com/elastic/package-registry/pull/392 locally and run this PR with `xpack.ingestManager.epm.registryUrl` set to `http://localhost:8080`, against another fresh elasticsearch database.
* Kibana should start without errors, login should work without errors.
* Verify in the Kibana dev console or with `curl` that after login, `GET _index_template` and `GET _component_template` correctly return all v2 index templates and component templates that were added / changed in https://github.com/elastic/package-registry/pull/392.
---
* Now verify (running against either package registry) that `GET _index_template` correctly returns a v2 index template for every dataset from the `system` package.
* Add a datasource for the `nginx` package, thereby triggering the installation of this package. This should work without error too. Again, verify with `GET _index_template` that index templates have been installed for all datasets of the package.
* In the UI, delete the newly created `nginx-1` datasource, then uninstall the `nginx` integration. Both operation should complete without errors.
* Verify with `GET _index_template` that the `nginx` index templates have been removed.
* Try to break it in any other way!
